### DISCUSSION TITLE
About `superfluous-else-raise` (**RET506**) and `elif`

### DISCUSSION THREAD

**Discussion by Anselmoo** [astral-sh/ruff#12468] [Q&A] [answered]
Currently, a combination of two `raise` cases connected via `if - elif` for **[RET506](https://docs.astral.sh/ruff/rules/superfluous-else-raise/)**? will raise an error.

```pass
 if not condition_1:
   	raise ...
 elif not if not condition_2:
	raise ...
# Here comes the actual code
```

Two separated `if`-blocks are fine.

```pass
 if not condition_1:
   	raise ...
 if not if not condition_2:
	raise ...
# Here comes the actual code
```

#### Why would it be a bad idea to allow `if—elif` for **RET506** when asking for a series of connected raise conditions?

In any case, thx `RUFF` for being the single source of linting!

**Comment by guillermodotn** [comment, marked as answer]
**RET506** is correct here.

If every branch of an if ends with `raise` (or `return/break/continue`), then the `else/elif` is unnecessary because the code after the block is only reachable when the condition wasn't met anyway.
